globus_sdk/auth/identity_map.py

```diff
@@ -0,0 +1,222 @@
+import re
+
+# what qualifies as a valid Identity Username?
+_USERNAME_PATTERN = re.compile(r"^[a-zA-Z0-9]+.*@[a-zA-z0-9-]+\..*[a-zA-Z]+$")
```

Comment:
I'm wondering if we shouldn't do this check from the other direction, i.e., check if the input is a UUID. That regex clearly matches things that aren't valid identity usernames and it's also complex enough that it conveys a false sense of security. I also worry that it might get reused somewhere else...

Anyway, I think what we really care about here is whether or not the input is a UUID so we know to treat it as an identity id--everything else gets dumped into the username bucket.

Reply:
Yeah, that seems very reasonable.
I think I pulled this regex out of globus-cli source (I'll dig it up later), so it's already "getting reused somewhere else".

I'll switch to a check that does `try: uuid.UUID(value) ; except ValueError: ...`